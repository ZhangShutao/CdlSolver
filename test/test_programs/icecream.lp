child(tommy).
ache(tommy).
% ache(tommy) | -ache(tommy).
teething(tommy).

%child(amy).

% the negation as failure in this program is used as a denial of a default rule
like(X) :- child(X), &c{like(X)}.
-like(X) :- ache(X), not teething(X), &c{-like(X)}.
:- not like(X), not -like(X), child(X).
