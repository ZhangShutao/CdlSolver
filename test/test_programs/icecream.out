default model: 1
x:ache(tommy),teething(tommy),like(tommy),child(tommy); y:ache(tommy),teething(tommy),child(tommy),like(tommy);
Satisfiable